course/models/models.py

```diff
@@ -101,6 +101,7 @@ def success_rate(self):
         return total_solved / total_tried
 
     def save(self, *args, **kwargs):
+        self.max_submission_allowed = 10 if self.event.type == 'EXAM' else 100
```

Comment:
This is in the wrong place. You have to specify it in the default attribute of the field. You can pass a function to it too (instead of a static value).

Reply:
That was my first approach, but I wasn't able to use a field in the model to set the default of another field until I had an instance of the model. See https://stackoverflow.com/questions/41343531/django-models-set-default-relative-to-another-field. Unless you know of another way to do it?

I can definitely write a function to set the number of attempts.

Reply:
Oh, I see. The problem is now if we change the max_submission_allowed for a question it will be overwritten and there will be no way for us to change that. What I recommend to do is to remove the default one we already have and in the save function set the default only if the max_submission_allowed is None.